Code/GraphMol/Substruct/SubstructUtils.cpp

```diff
@@ -163,6 +186,9 @@ bool bondCompat(const Bond *b1, const Bond *b2,
       res = false;
     }
   }
+  if (res) {
```

Comment:
same comment about avoiding the call if not needed
```suggestion
  if (res && !ps.bondProperties.empty()) {
```